push/sender/src/main/java/org/jboss/aerogear/unifiedpush/message/sender/AdmPushNotificationSender.java

```diff
@@ -0,0 +1,63 @@
+package org.jboss.aerogear.unifiedpush.message.sender;
+
+
+import org.jboss.aerogear.unifiedpush.api.AdmVariant;
+import org.jboss.aerogear.unifiedpush.api.Variant;
+
+import org.jboss.aerogear.unifiedpush.message.UnifiedPushMessage;
+import org.jboss.aerogear.unifiedpush.utils.AeroGearLogger;
+import org.jboss.aerogear.adm.ADM;
+import org.jboss.aerogear.adm.AdmService;
+import org.jboss.aerogear.adm.PayloadBuilder;
+
+import java.util.Collection;
+import java.util.Set;
+
+@SenderType(AdmVariant.class)
+public class AdmPushNotificationSender implements PushNotificationSender {
+    private final AeroGearLogger logger = AeroGearLogger.getInstance(AdmPushNotificationSender.class);
+
+    @Override
+    public void sendPushMessage(Variant variant, Collection<String> clientIdentifiers, UnifiedPushMessage pushMessage, NotificationSenderCallback senderCallback) {
+        AdmService admService = ADM.newService();
+
+        PayloadBuilder builder = ADM.newPayload();
+
+        //flatten the "special keys"
+        builder.dataField("alert", pushMessage.getMessage().getAlert());
+        builder.dataField("sound", pushMessage.getMessage().getSound());
+        builder.dataField("badge", "" + pushMessage.getMessage().getBadge());
```

Comment:
in here as well https://aerogear.org/docs/unifiedpush/push-message-format/

On Mon, Feb 9, 2015 at 3:54 PM, Sebastien Blanc <notifications@github.com>
wrote:

> In
> push/sender/src/main/java/org/jboss/aerogear/unifiedpush/message/sender/AdmPushNotificationSender.java
> <https://github.com/aerogear/aerogear-unifiedpush-server/pull/480#discussion_r24332632>
> :
>
> > +import java.util.Set;
> > +
> > +@SenderType(AdmVariant.class)
> > +public class AdmPushNotificationSender implements PushNotificationSender {
> > +    private final AeroGearLogger logger = AeroGearLogger.getInstance(AdmPushNotificationSender.class);
> > +
> > +    @Override
> > +    public void sendPushMessage(Variant variant, Collection<String> clientIdentifiers, UnifiedPushMessage pushMessage, NotificationSenderCallback senderCallback) {
> > +        AdmService admService = ADM.newService();
> > +
> > +        PayloadBuilder builder = ADM.newPayload();
> > +
> > +        //flatten the "special keys"
> > +        builder.dataField("alert", pushMessage.getMessage().getAlert());
> > +        builder.dataField("sound", pushMessage.getMessage().getSound());
> > +        builder.dataField("badge", "" + pushMessage.getMessage().getBadge());
>
> Yes, let's keep alert also for ADM. By documentating, you mean in the
> client SDK and also in the Sender (Java and NodeJS) ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/aerogear/aerogear-unifiedpush-server/pull/480/files#r24332632>
> .
>



-- 
Matthias Wessendorf

blog: http://matthiaswessendorf.wordpress.com/
sessions: http://www.slideshare.net/mwessendorf
twitter: http://twitter.com/mwessendorf